### PR TITLE
Overmap Flags System

### PR DESCRIPTION
**Pull request by Andromeda-K22**
Prequisite for Overmap Combat.

### CODE REVIEW

File: maps/torch/job/torch_jobs.dm [hunk 1]
```diff
@@ -18,8 +18,6 @@
 		/datum/species/machine = list(HUMAN_ONLY_JOBS, /datum/job/representative),
 		/datum/species/diona   = list(HUMAN_ONLY_JOBS, /datum/job/captain, /datum/job/hop, /datum/job/hos, /datum/job/officer, /datum/job/warden),
 		/datum/species/sergal = list(HUMAN_ONLY_JOBS, /datum/job/representative),
-		/datum/species/sergal/northern = list(HUMAN_ONLY_JOBS, /datum/job/representative),
-		/datum/species/sergal/eastern = list(HUMAN_ONLY_JOBS, /datum/job/representative),
```
Comment (Roland410) [inline maps/torch/job/torch_jobs.dm]: Probably not intended?

Comment (Andromeda-K22) [reply maps/torch/job/torch_jobs.dm]: Unintentional but northern/eastern were already removed - shouldn't be a problem.

File: code/__defines/flags.dm [hunk 2]
```diff
@@ -49,3 +49,8 @@ GLOBAL_LIST_INIT(bitflags, list(1, 2, 4, 8, 16, 32, 64, 128, 256, 512, 1024, 204
 #define PASS_FLAG_TABLE  0x1
 #define PASS_FLAG_GLASS  0x2
 #define PASS_FLAG_GRILLE 0x4
+
+#define OVERMAP_SECTOR_BASE              0x0001 // Whether or not this sector is a starting sector. Z levels contained in this sector are added to station_levels
```
Comment (MarinaGryphon) [inline code/__defines/flags.dm]: Minor nitpick, but I think `1 >> 0`, `1 >> 1`, `1 >> 2` are easier for bitfields than putting powers of two in manuall.

File: code/__defines/flags.dm [hunk 3]
```diff
@@ -49,3 +49,8 @@ GLOBAL_LIST_INIT(bitflags, list(1, 2, 4, 8, 16, 32, 64, 128, 256, 512, 1024, 204
...
+#define OVERMAP_SECTOR_KNOWN             0x0002 // Makes the sector show up on nav computers
+#define OVERMAP_SECTOR_IN_SPACE          0x0004 // If the sector can be accessed by drifting off the map edge
+#define OVERMAP_SECTOR_UNTARGETABLE      0x0008 // If the sector is untargetable by missiles
```
Comment (MarinaGryphon) [inline code/__defines/flags.dm]: missiles, owo

File: maps/away/skrellscoutship/skrellscoutship.dm [hunk 4]
```diff
@@ -20,7 +20,7 @@
 /obj/effect/overmap/visitable/sector/skrellscoutspace
 	name = "Empty Sector"
 	desc = "Slight traces of a cloaking device are present. Unable to determine exact location."
-	in_space = 1
+	sector_flags = OVERMAP_SECTOR_IN_SPACE
```
Comment (MarinaGryphon) [inline maps/away/skrellscoutship/skrellscoutship.dm]: Not visible due to lacking `OVERMAP_SECTOR_KNOWN`

Comment (MarinaGryphon) [reply maps/away/skrellscoutship/skrellscoutship.dm]: oh wait this flag isn't used anywhere yet iirc and is just for the future stuff aight

**Pull request merged**